Gems/Atom/Feature/Common/Code/Source/FrameCaptureSystemComponent.cpp

```diff
@@ -446,27 +488,29 @@ namespace AZ
                 capture->m_outputFilePath = ResolvePath(imagePath);
             }
 
-            return captureHandle;
+            return AZ::Success(captureHandle);
         }
 
-        FrameCaptureId FrameCaptureSystemComponent::CaptureScreenshotForWindow(const AZStd::string& filePath, AzFramework::NativeWindowHandle windowHandle)
+        AZ::Outcome<FrameCaptureId, FrameCaptureError> FrameCaptureSystemComponent::CaptureScreenshotForWindow(const AZStd::string& filePath, AzFramework::NativeWindowHandle windowHandle)
         {
             return InternalCaptureScreenshot(filePath, windowHandle);
         }
 
-        FrameCaptureId FrameCaptureSystemComponent::CaptureScreenshot(const AZStd::string& filePath)
+        AZ::Outcome<FrameCaptureId, FrameCaptureError> FrameCaptureSystemComponent::CaptureScreenshot(const AZStd::string& filePath)
         {
             AzFramework::NativeWindowHandle windowHandle = AZ::RPI::ViewportContextRequests::Get()->GetDefaultViewportContext()->GetWindowHandle();
-            if (windowHandle)
+            if (windowHandle == nullptr)
             {
-                return InternalCaptureScreenshot(filePath, windowHandle);
+                return InvalidFrameCaptureId;
```

Comment:
Should be AZ::Failure?

Reply:
Nice catch. Compilers won't warn for this.